pkg/cmd/server/server.go

```diff
@@ -59,9 +59,12 @@ var ServerCmd = &cobra.Command{
 		select {
 		case err := <-errCh:
 			log.Fatal(err)
-		// TODO: This is an optimistic check. We should check if the server is actually running
-		case <-time.After(5 * time.Second):
-			printServerStartedMessage(c)
+		case <-time.After(1 * time.Second):
```

Comment:
This again is not ideal as the error might get sent while the health check is happening and it would again be suppressed.

I suggest you put the error channel check in a go routine and then do the health check below it.